### ISSUE TITLE
Cleanup after failed execution

### ISSUE THREAD

**Issue by teebu** [alexcasalboni/aws-lambda-power-tuning#10]
I noticed if a lambda is long (200 seconds) running it in default non parallel mode, will cause the executioner to time out, because it will run it in sequence. 

note: documentation is wrong, the parameter is `parallelInvocation` not `enableParallel`

The failure results in executioner exceeding 300sec run time after the 2nd invocation. If you set `num` to 10, it will take 200 * 10 seconds to finish which it will never do.

In case of failure, it should still clean up the aliases/versions it created.

And the error message is unknown, it should say that the executioner timed out, or something to indicate that the person should run these in parallel mode. 

```
Error

Lambda.Unknown
Cause

The cause could not be determined because Lambda did not return an error type.
```

Side note, would be nice to finish the rest and not cancel on first failed. If you have an array of lambdas 128 to whatever, and 128 is too small, it breaks the testing for the rest of them.

**Comment by alexcasalboni** [comment]
Hi @teebu.

Let's take one problem at a time:

1. The documentation is wrong. It should be `parallelInvocation`. Easy to fix :)
2. You are right, the total execution time (either parallel or serial) should stay below 300 seconds. Which means that if you go for serial invocations, the average duration of your functions should be below 3 seconds with `num=100`, 30 seconds with `num=10`, and so on. This should probably be clearer in the documentation, but I can't think of any simple workaround besides setting `num` accordingly.
3. The `cleaner` function is currently not executed if something goes wrong with the `executor` function. To fix this, we'd need to complicate the state machine and add some advanced error handling.
4. Such error handling could also make each branch (i.e. "power branch") independent, so that you can gather useful data even if one of the branches fails.


These are all great thoughts and ideas :) 

I'm fixing point 1 asap 🚀 

Point 2 is hopeless, imho 😢 

Point 3 would be a great enhancement! 🎉 Would you have time to work on it?

Regarding point 4, my initial idea is that if something crashes you won't get much meaningful information so it was totally ok for me to fail the whole execution (and eventually spare money & time). That said, if we solve point 3 we should be good with point 4 as well!

**Comment by alexcasalboni** [comment]
I have implemented error handling in #27 :)

Now all the aliases and versions are cleaned up even in case of error.
